{
  "className": "qx.core.AssertionError",
  "packageName": "qx.core",
  "name": "AssertionError",
  "superClass": "qx.type.BaseError",
  "interfaces": [],
  "mixins": [],
  "functionName": null,
  "clazz": {
    "location": {
      "start": {
        "line": 23,
        "column": 0,
        "index": 568
      },
      "end": {
        "line": 59,
        "column": 3,
        "index": 1553
      }
    },
    "jsdoc": {
      "@description": [
        {
          "name": "@description",
          "body": "<p>Assertion errors are thrown if an assertion in {@link qx.core.Assert}\n fails.</p>"
        }
      ]
    }
  },
  "type": "class",
  "construct": {
    "location": {
      "start": {
        "line": 36,
        "column": 2,
        "index": 986
      },
      "end": {
        "line": 39,
        "column": 3,
        "index": 1133
      }
    },
    "jsdoc": {
      "@description": [
        {
          "name": "@description",
          "body": ""
        }
      ],
      "@param": [
        {
          "name": "@param",
          "body": "comment {String} Comment passed to the assertion call",
          "paramName": "comment",
          "description": " Comment passed to the assertion call",
          "type": "String"
        },
        {
          "name": "@param",
          "body": "failMessage {String} Fail message provided by the assertion",
          "paramName": "failMessage",
          "description": " Fail message provided by the assertion",
          "type": "String"
        }
      ]
    }
  },
  "members": {
    "__trace": {
      "location": {
        "start": {
          "line": 48,
          "column": 4,
          "index": 1337
        },
        "end": {
          "line": 48,
          "column": 17,
          "index": 1350
        }
      },
      "type": "variable",
      "access": "private"
    },
    "getStackTrace": {
      "location": {
        "start": {
          "line": 55,
          "column": 4,
          "index": 1495
        },
        "end": {
          "line": 57,
          "column": 5,
          "index": 1545
        }
      },
      "jsdoc": {
        "@description": [
          {
            "name": "@description",
            "body": "<p>Stack trace of the error</p>"
          }
        ],
        "@return": [
          {
            "name": "@return",
            "body": "{String[]} The stack trace of the location the exception was thrown",
            "type": "String[]",
            "description": " The stack trace of the location the exception was thrown"
          }
        ]
      },
      "type": "function",
      "access": "public"
    }
  },
  "descendants": []
}